src/components/PeopleManagement/GroupDetailPage.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { useIntl, FormattedMessage } from '@edx/frontend-platform/i18n';
+import {
+  Breadcrumb, Card, Hyperlink, Icon, IconButton, IconButtonWithTooltip, Toast, useToggle,
+} from '@openedx/paragon';
+import { Delete, Edit } from '@openedx/paragon/icons';
+
+import { useEnterpriseGroupUuid } from '../learner-credit-management/data';
+import { ROUTE_NAMES } from '../EnterpriseApp/data/constants';
+import DeleteGroupModal from './DeleteGroupModal';
+import EditGroupNameModal from './EditGroupNameModal';
+
+const GroupDetailPage = () => {
+  const intl = useIntl();
+  const { enterpriseSlug, groupUuid } = useParams();
+  const { data: enterpriseGroup } = useEnterpriseGroupUuid(groupUuid);
+  const [isDeleteModalOpen, openDeleteModal, closeDeleteModal] = useToggle(false);
+  const [isEditModalOpen, openEditModal, closeEditModal] = useToggle(false);
+  const [showToast, setShowToast] = useState(false);
+  const [toastMessage, setToastMessage] = useState('');
+
+  const tooltipContent = (
+    <FormattedMessage
+      id="adminPortal.peopleManagement.groupDetail.deleteGroup.icon"
+      defaultMessage="Delete group"
+      description="Tooltip to confirm meaning of trash icon."
+    />
+  );
+
+  const budgetNameAndEdit = (
+    <>
+      <span className="pr-1">{enterpriseGroup?.name}</span>
+      <IconButton
+        key="editGroupTooltip"
+        src={Edit}
+        iconAs={Icon}
+        alt="Edit group"
+        className="mr-2"
+        onClick={openEditModal}
+        size="sm"
+        data-testid="edit-modal-icon"
+      />
+    </>
+  );
+  return (
+    <div className="pt-4 pl-4">
+      <Toast
+        onClose={() => setShowToast(false)}
+        show={showToast}
+      >
+        {toastMessage}
+      </Toast>
+      <DeleteGroupModal
+        group={enterpriseGroup}
+        isOpen={isDeleteModalOpen}
+        close={closeDeleteModal}
+      />
+      <EditGroupNameModal
+        group={enterpriseGroup}
+        isOpen={isEditModalOpen}
+        close={closeEditModal}
+        setShowToast={setShowToast}
+        setToastMessage={setToastMessage}
+      />
+      <Breadcrumb
+        aria-label="people management breadcrumb navigation"
+        links={[
+          {
+            label: intl.formatMessage({
+              id: 'adminPortal.peopleManagement.breadcrumb',
+              defaultMessage: 'People Management',
+              description: 'Breadcrumb label for the people management page',
+            }),
+            href: `/${enterpriseSlug}/admin/${ROUTE_NAMES.peopleManagement}`,
+          },
+        ]}
+        activeLabel={enterpriseGroup?.name}
+      />
+
+      <Card orientation="horizontal">
+        <Card.Body>
+          <Card.Header
+            title={budgetNameAndEdit}
+            subtitle={`${enterpriseGroup?.acceptedMembersCount} accepted members`}
```

Comment:
I think we should avoid this conditional check by adding in a loading state and `useEffect` hook. i'm seeing this first say `undefined` in the UI and then the count populates (see screen recording). 

To guarantee that the `enterpriseGroup` data exists before rendering the card, we can do something like the sample code below. When the loading state is true, we could render the `Skeleton`.

>  const [isLoading, setIsLoading] = useState(true);
> 
>   useEffect(() => {
>     if (enterpriseGroup !== undefined) {
>       setIsLoading(false);
>     }
>   }, [enterpriseGroup]);

https://github.com/user-attachments/assets/5118fb1a-f4d2-4805-8937-ab0be94548ac